### DISCUSSION TITLE
Stuck building the container.

### DISCUSSION THREAD

**Discussion by Dgreendow** [toverainc/willow#153] [Q&A]
Hello everyone. 
I am not well versed in docker and maybe I’m in over my head but would love to get this up and running. There is something very obvious that I am missing while trying to build the container. Is there a command that I need to run before /utils.sh build-docker? I’ve tried every command that I can think of. Am I supposed to be in a certain directory while running this?? 

I am running on a raspberry pi 4 running Pi OS. 

Any help would be appreciated.

**Comment by nikito** [comment]
Do you have docker installed? It's mentioned docker is used to build the container, the scripts don't install it for you though, you'd need to install that first before running the script. 🙂

**Reply by Dgreendow** [reply]
Yes, docker is installed and running. 
If I run the script from the willow directory ( should I even be running it there?), I get this error. 

 => [internal] load .dockerignore                                               0.0s
 => => transferring context: 2B                                                 0.0s
 => [internal] load build definition from Dockerfile                            0.1s
 => => transferring dockerfile: 770B                                            0.0s
 => ERROR [internal] load metadata for docker.io/espressif/idf:release-v4.4     0.8s
------
 > [internal] load metadata for docker.io/espressif/idf:release-v4.4:
------
Dockerfile:1
--------------------
   1 | >>> FROM espressif/idf:release-v4.4
   2 |     
   3 |     ARG DEBIAN_FRONTEND="noninteractive"
--------------------
ERROR: failed to solve: espressif/idf:release-v4.4: no match for platform in manifest sha256:2766db6ad8760d3c4ac52068367aff9be7c82f598025481023816d0112d9eb68: not found


I’ll have to do a little digging when I have a chance to think for more than 20 seconds  (work,kids,dogs…) but it looks like this may not work with Debian?

**Reply by nikito** [reply]
Hm, it may be that because you are running on an ARM processor the docker container is incompatible?

**Reply by Dgreendow** [reply]
It very well could be. When I get a chance, I will attempt to spin up docker on proxmox and see if that makes a difference. The hope was that someone else had already ran into this and could guide me a little. I’m relatively new to this world and learning as I go. I appreciate the suggestions.

**Reply by nikito** [reply]
I have mine spun up in an Ubuntu vm on proxmox and didn't run into this issue, pretty sure debian should work as well. Don't think I've seen anyone try to run on an rpi unfortunately. 🫤

**Reply by Dgreendow** [reply]
Good to know! I’ll give that a shot when I get a chance. I’m new to proxmox, so I tried the pi solely for familiarity and easy access to the cli from my phone when I have a few minutes here and there. Thank you again, much appreciated.

**Reply by kristiankielhofner** [reply]
If you look at the [tags](https://hub.docker.com/r/espressif/idf/tags) for the Espressif IDF container you will see that they are all amd64 only. So even if we wanted to try to support ARM we would run into issues with our BASE container which doesn't have support for ARM.

I've noted it elsewhere but we're currently in the process of moving most of the build-time configuration to dynamic configuration (via API), implementing a web based flashing mechanism, etc to make the current build process only required for Willow development.

This will be our 1.0 release and many/most users will be able to take an ESP BOX out of the box and flash and configure Willow with a few clicks from a web browser. Over the air updates will be supported as well.

**Reply by ghost** [reply]
work fine fine on PI 4 do the following.

Install latest Raspberry Pi OS Lite (64 bit)

make sure you enable ssh 

login via ssh 
 
sudo apt update
sudo apt upgrade
apt-get install python3-venv
curl -sSL https://get.docker.com/ | sh
sudo usermod -aG docker $USER
logout

Log back in and follow the steps carefully in the github . Adding the current user into the docker group , will ensure you will not need to add the sudo command.
